app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,45 @@
+class RestaurantsController < ApplicationController
+   def index
+     @restaurant = Restaurant.all
+   end
+ 
+   def show
+     @restaurant = Restaurant.find(params[:id])
+   end
+ 
+   def new
+     @restaurant = Restaurant.new
+   end
+ 
+   def edit
+     @restaurant = Restaurant.find(params[:id])
+   end
+ 
+   def create
+     @restaurant = Restaurant.create(restaurant_params)
+     if @restaurant.save 
+       redirect_to root_path
+       flash[:notice] = "Restaurant  was  saved  successfully"
+     else 

```

Comment:
🔨 @joni88  The directions called for the "happy path" approach (i.e. there will be no errors, there are no validation rules set yet), which means this conditional is not needed since.  Error handling and Flash messages are part of HW for Lesson 10 (not this hw assignment)

"Happy Path" for the create action looks like this:

```

def create
  @restaurant = Restaurant.create(restaurant_params)
  redirect_to characters_path 
end
```